src/message/pack_encrypted/mod.rs

```diff
@@ -129,6 +134,72 @@ impl Message {
 
         Ok((msg, metadata))
     }
+
+    fn validate(&self, to: &str, from: Option<&str>, sign_by: Option<&str>) -> Result<bool> {
+        if !is_did(to) {
+            Err(err_msg(
+                ErrorKind::IllegalArgument,
+                format!("`to` value is not a valid DID of DID URL: {}", to),
+            ))?;
+        }
+
+        match from {
+            Some(from) if !is_did(from) => Err(err_msg(
+                ErrorKind::IllegalArgument,
+                format!("`from` value is not a valid DID of DID URL: {}", from),
+            ))?,
+            _ => {}
+        }
+
+        match sign_by {
+            Some(sign_by) if !is_did(sign_by) => Err(err_msg(
+                ErrorKind::IllegalArgument,
+                format!(
+                    "`sign_from` value is not a valid DID of DID URL: {}",
+                    sign_by
+                ),
+            ))?,
+            _ => {}
+        }
+
+        let (to_did, _) = did_or_url(to);
+
+        let self_to = &self.to;
+        match self_to {
+            Some(self_to) if !self_to.contains(&to_did.into()) => {
+                Err(err_msg(
+                    ErrorKind::IllegalArgument,
+                    format!(
+                        "`message.to` value {:?} does not contain `to` value's DID {}",
+                        self_to, to_did
+                    ),
+                ))?;
+            }
+            _ => {}
+        }
+
+        match from {
```

Comment:
I think code will be much more straightforward if you will use matching on all input data

```
match (from, self.from) {
  (Some(ref from), Some(ref sfrom)) if did_or_url(from).0 != sfrom,
  Some(_), None, errr,
  None, Some(_), err,
}
```

Reply:
Similar to other matches